### ISSUE TITLE
Add lfs data to private repo used in testing

### ISSUE THREAD

**Issue by ableuler** [SwissDataScienceCenter/renku#1409]
The test of launching a notebook from a private repos should include some lfs data to capture things like https://github.com/SwissDataScienceCenter/renku-notebooks/issues/375.

**Comment by rokroskar** [comment]
Do we have a test for launching notebooks from a private repo? afaik we don't

**Comment by rokroskar** [comment]
The other thing that would have caught SwissDataScienceCenter/renku-notebooks#375 is running acceptance tests after a renkulab upgrade. Afaik we never run acceptance tests on prod... 🤔

**Comment by ableuler** [comment]
> The other thing that would have caught [SwissDataScienceCenter/renku-notebooks#375](https://github.com/SwissDataScienceCenter/renku-notebooks/issues/375) is running acceptance tests after a renkulab upgrade. Afaik we never run acceptance tests on prod... 🤔

Yeah, even if it might not have caught this bug, it would be good practice I think. Has there been any recent attempt to run our tests through helm?

**Comment by pameladelgado** [comment]
> Afaik we never run acceptance tests on prod...

Yes, that's the case. We could automate this for our deployments, even if some issue is caught 'after the fact' which is what we try to avoid by running the tests on dev.
Also, if we are going to use these tests for prod deployments, at some point we will probably need to take care of versioning the tests to keep them consistent with the features present in a given release. It happens that a release with a certain feature is made and the tests for that feature are done some time after or that we come up with better ways of testing existing features (like this specific case).

> Has there been any recent attempt to run our tests through helm?

I dont think so. 
I agree that it would be nice to have the possibility to run tests via helm and I think that we should make it generic enough for testing external/new deployments.

**Comment by rokroskar** [comment]
We used to run tests through helm. There are some complications with the current set up iirc. 

> Also, if we are going to use these tests for prod deployments, at some point we will probably need to take care of versioning the tests to keep them consistent with the features present in a given release. 

what do you mean? The tests _are_ versioned in the same sense that the releases/renku is versioned. If there aren't tests for features we release then that means we need an improvement to the test process so that this doesn't happen.

**Comment by pameladelgado** [comment]
> what do you mean? The tests _are_ versioned in the same sense that the releases/renku is versioned. If there aren't tests for features we release then that means we need an improvement to the test process so that this doesn't happen.

yes, I mean that we either: ship the needed tests needed for a given release and its features at the same time as the renku release (improving our test process) or make test releases/versioning separately. For Helm tests the first option makes more sense, but our current tests are kind of independent from the charts so the second option is possible. However, I think in general is desirable to not have to maintain two releases/versions.

**Comment by ciyer** [comment]
We do strive to ship the tests together with the new features as part of a release. The exception to this is if we know that the current presentation of a feature is temporary or incomplete; then we will hold off implementing tests until the UI stabilizes.
